coding.md

```diff
@@ -1,108 +1,220 @@
+
+## :heavy_check_mark: [Write code for humans](http://douglasorr.github.io/2020-03-data-for-machines/article.html)
+> "Code is read more often than it is written." — [Guido van Rossum](https://twitter.com/gvanrossum) (creator of Python)
+
+> "Programs are meant to be read by humans and only incidentally for computers to execute."  
+> — Donald Knuth, The Art of Computer Programming
+
+Writing code that humans can understand is an investment to have maintainable and reusable code. If it is easy to read, maintenance and  edits will be much quicker in the future and it will be much easier for other people to work with your code. 
+The shortest, most efficient code for the computer is likely not the optimal code for human readability. 
+
+### Naming conventions
+Proper naming conventions will help collaboration in big teams and it makes the code easier to maintain. The three most widely-used conventions amongst programming communities are:
+* camelCase: Names start with a small letter and every subsequent will start with upperCase
+* PascalCase: Similar to camelCase but the first letter of the string is also UpperCase
+* snake_case: All names are lower case with an underscore between words
+
+Whichever convention you choose, make sure to stick with it for the duration of your project.
+
+### Maintain meaningful and proper object names
+  
+`height_in_metres()` is better than `converter()` for a function that converts height into metres. This makes it more obvious what your code is doing and makes your code more readable
+
+### Maintain meaningful and proper column names
+Most of the data we read is from either an Excel file or a poorly designed database, thus we see column names with spaces and dots and other rogue characters. Remember these rules for column names:
+
+1. Stick to consistent naming convention (see above)
+2. Assume everything is case-sensitive.
+3. Do not use special characters ever.
+4. Do not add spaces.
+
+### Do not use numbers to specify columns
+In R you can use numbers to refer to columns, but (to parapharse a noted scholar) just because you can doesn’t mean you should.
+
+   #### Bad:
+
+    ```
+    # In base R
+    mtcars[,c(1,3:5,8)]
+
+    # Using tidyverse
+    mtcars %>% select(1, 3:5, 8)
+    ```
+
+   #### Good:
+
+    ```
+    # In base R
+    mtcars[, c("mpg", "disp", "hp", "drat", "vs")]
+
+    # Using tidyverse
+    mtcars %>% 
+      select(mpg, disp, hp, drat, vs)
+    ```
+
+### Conditional statements
+
+#### Do
+* Use `case_when()` within the `tidyverse` collection for conditional statements. This is a more readable way of dealing with many ifelse statements. Standard `if_else` statements can be very useful but can become hard to understand if too many are used.
+#### Don't
+* Use deeply nested `for` loops or `if_else` statements
+
+ 
+## :heavy_check_mark: Use a style guide 
+The default for Scottish Government is the [Tidyverse styleguide](https://style.tidyverse.org/)  
```

Comment:
An example of what I mean in my general comments on making stronger statements, I think this could be:

In the Scottish Government, R code should (must?) be styled following the [Tidyverse styleguide](https://style.tidyverse.org/).